organization.go

```diff
@@ -439,19 +447,53 @@ func (s *organizations) ReadRunQueue(ctx context.Context, organization string, o
 	return rq, nil
 }
 
-func (s *organizations) ReadDataRetentionPolicy(ctx context.Context, organization string) (*DataRetentionPolicy, error) {
+func (s *organizations) ReadDataRetentionPolicy(ctx context.Context, organization string) (*DataRetentionPolicyChoice, error) {
 	if !validStringID(&organization) {
 		return nil, ErrInvalidOrg
 	}
 
+	// The API to read the drp is org/<name>/relationships/data-retention-policy
+	// However, this API can return multiple "types" (e.g. data-retention-policy-delete-olders, or data-retention-policy-don-deletes)
+	// Ideally we would deserialize this directly into the choice type (DataRetentionPolicyChoice)...however, there isn't a way to
+	// tell the current jsonapi implementation that the direct result of an endpoint could be different types. Relationships can be polymorphic,
+	// but the direct result of an endpoint can't be (as far as the jsonapi implementation is concerned)
+
+	// Instead, we need to figure out the type of the data retention policy first, and deserialize it into the matching model. We
+	// can then create a choice type manually
+	org, err := s.Read(ctx, organization)
+	if err != nil {
+		return nil, err
+	}
+
+	// there is no drp (of a known type)
+	if org.DataRetentionPolicy == nil || !org.DataRetentionPolicy.IsPopulated() {
+		return org.DataRetentionPolicy, nil
```

Comment:
I am not sure if this should be like this, where we return a nil or empty `*DataRetentionPolicyChoice` if the org has no DRP....or if we should return a `ResourceNotFound` error. Open to feedback